### DISCUSSION TITLE
How can I get user custom attribute value in property mapping?

### DISCUSSION THREAD

**Discussion by ser** [goauthentik/authentik#12485] [Q&A]
My users have set a custom attribute 'upn'

![image](https://github.com/user-attachments/assets/bd337d06-521d-4881-b94d-ff8ee8416990)

How can I get this attribute in property mapping? The example property mapping for proxy is:

```
return {
    "ak_proxy": {
        "user_attributes": {
            "additionalHeaders": {
                "X-App-User": request.user.username
            }
        }
    }
}
```
... and I really do not get how can I call that `upn` attribute instead of `request.user.username` in above example. I tried `user.attributes.upn` but Mr Authentik says there is no such thing.

Please help!

**Comment by rissson** [comment]
`request.user.attributes["upn"]`
